specs/span-batches.md

```diff
@@ -0,0 +1,203 @@
+  - Note: The sequencing window is enforced for the *batch as a whole*:
+    if the batch was partially invalid instead, it would drop the oldest L2 blocks,
+    which makes the later L2 blocks invalid.
+  - Variables:
+    - `origin_changed_bit = origin_bits[0]`: `true` if the first L2 block changed its L1 origin, `false` otherwise.
+    - `start_epoch_num = safe_l2_head.origin.block_number + (origin_changed_bit ? 1 : 0)`
+    - `end_epoch_num = safe_l2_head.origin.block_number + sum(origin_bits)`: block number of last referenced L1 origin
+  - Rules:
+    - `start_epoch_num + sequence_window_size < inclusion_block_number` -> `drop`:
+      i.e. the batch must be included timely.
+    - `end_epoch_num < epoch.number` -> `future`: i.e. all referenced L1 epochs must be there.
+    - `end_epoch_num == epoch.number`:
+      - If `batch.l1_origin_check != epoch.hash[:20]` -> `drop`: verify the batch is intended for this L1 chain.
+    - `end_epoch_num > epoch.number` -> `drop`: must have been duplicate batch,
+      we may be past this L1 block in the safe L2 chain.
+- Max Sequencer time-drift checks:
+  - Note: The max time-drift is enforced for the *batch as a whole*, to keep the possible output variants small.
+  - Variables:
+    - `block_input`: an L2 block from the span-batch,
+      with L1 origin as derived from the `origin_bits` and now established canonical L1 chain.
+    - `next_epoch` is relative to the `block_input`,
+      and may reach to the next origin outside of the L1 origins of the span.
+  - Rules:
+    - For each `block_input` that can be read from the span-batch:
+      - `block_input.timestamp < block_input.origin.time` -> `drop`: enforce the min L2 timestamp rule.
+      - `block_input.timestamp > block_input.origin.time + max_sequencer_drift`: enforce the L2 timestamp drift rule,
+        but with exceptions to preserve above min L2 timestamp invariant:
+        - `len(block_input.transactions) == 0`:
+          - `epoch.number == batch.epoch_num`:
+            this implies the batch does not already advance the L1 origin,
+            and must thus be checked against `next_epoch`.
+            - If `next_epoch` is not known -> `undecided`:
+              without the next L1 origin we cannot yet determine if time invariant could have been kept.
+            - If `batch.timestamp >= next_epoch.time` -> `drop`:
+              the batch could have adopted the next L1 origin without breaking the `L2 time >= L1 time` invariant.
+        - `len(batch.transactions) > 0`: -> `drop`:
+          when exceeding the sequencer time drift, never allow the sequencer to include transactions.
+- And for all transactions:
+  - `drop` if the `batch.transactions` list contains a transaction
+    that is invalid or derived by other means exclusively:
+    - any transaction that is empty (zero length `tx_data`)
+    - any [deposited transactions][g-deposit-tx-type] (identified by the transaction type prefix byte in `tx_data`)
```

Comment:
I think the `g-deposit-tx-type` inline link is missing

Reply:
fixed, thank you